### ISSUE TITLE
Higher level API for simple cases

### ISSUE THREAD

**Issue by cmbruns** [cmbruns/pyopenxr#71]
```python
import xr

with xr.something() as something:
  my_initialization_code()
  while xr.some_loop_thing():
    my_per_frame_code()
```

**Comment by cmbruns** [comment]
or even eventually

```python
import xr

for something in xr.loop.headless():
  my_tracker_logger_or_whatever(something)
```

**Comment by cmbruns** [comment]
see use of `xr.ContextObject(...)` in examples repo.
Fixed for 1.0.2401
